src/librustc_driver/driver.rs

```diff
@@ -1247,7 +1249,9 @@ pub fn collect_crate_types(session: &Session, attrs: &[ast::Attribute]) -> Vec<c
 }
 
 pub fn compute_crate_disambiguator(session: &Session) -> String {
-    let mut hasher = Sha256::new();
+    use std::hash::Hasher;
+
+    let mut hasher = ArchIndependentHasher::new(Blake2bHasher::new(256 / 8, &[]));
```

Comment:
I shared this patch with one of the blake2 designers and he was concerned about the use of 128 bits. See this titter thread: https://twitter.com/zooko/status/794972598853627904?s=09